deployment/build-s3-dist.sh

```diff
@@ -0,0 +1,441 @@
+# Build Functions 
+#------------------------------------------------------------------------------
+# Echo, execute, and check the return code for a command. Exit if rc > 0
+# ex. do_cmd npm run build
+usage() 
+{
+    echo "Usage: $0 bucket solution-name version"
+    echo "Please provide the base source bucket name, trademarked solution name, and version." 
+    echo "For example: ./build-s3-dist.sh mybucket my-solution v1.0.0" 
+    exit 1 
+}
+
+do_cmd() 
+{
+    echo "------ EXEC $*"
+    $*
+    rc=$?
+    if [ $rc -gt 0 ]
+    then
+            echo "Aborted - rc=$rc"
+            exit $rc
+    fi
+}
+
+sedi()
+{
+    # cross-platform for sed -i
+    sed -i $* 2>/dev/null || sed -i "" $*
+}
+
+# use sed to perform token replacement
+# ex. do_replace myfile.json %%VERSION%% v1.1.1
+do_replace() 
+{
+    replace="s/$2/$3/g"
+    file=$1
+    do_cmd sedi $replace $file
+}
+
+create_template_json() 
+{
+    # Run 'cdk synth' to generate raw solution outputs
+    do_cmd rushx cdk context --clear && STAGE=$STAGE rushx cdk synth -q --output=$staging_dist_dir
+
+    echo "Removing staging directory, if exists"
+    # Remove unnecessary output files
+    do_cmd cd $staging_dist_dir
+    # ignore return code - can be non-zero if any of these does not exist
+    echo "Removing output files if they exist"
+    rm tree.json manifest.json cdk.out
+
+    echo "Check folders in staging_dist_dir (<codebuildID>/src/deployment/staging)"
+    do_cmd ls -l $staging_dist_dir
+    echo "Check all contents of staging_dist_dir (<codebuildID>/src/deployment/staging)"
+    do_cmd ls -R $staging_dist_dir
+
+    # Move outputs from staging to template_dist_dir
+    echo "Move outputs from staging to template_dist_dir"
+    do_cmd mv $staging_dist_dir/*.template.json $template_dist_dir/
+
+    # Rename all *.template.json files to *.template
+    echo "Rename all *.template.json to *.template"
+    echo "copy templates and rename"
+    for f in $template_dist_dir/*.template.json; do
+        mv -- "$f" "${f%.template.json}.template"
+    done
+}
+
+create_template_yaml() 
+{
+    # Assumes current working directory is where the CDK is defined
+    # Output YAML - this is currently the only way to do this for multiple templates
+    maxrc=0
+    for template in `cdk list`; do
+        echo Create template $template
+        cdk synth $template > ${template_dist_dir}/${template}.template
+        if [[ $? > $maxrc ]]; then
+            maxrc=$?
+        fi
+    done
+}
+
+cleanup_temporary_generted_files()
+{
+    echo "------------------------------------------------------------------------------"
+    echo "${bold}[Cleanup] Remove temporary files${normal}"
+    echo "------------------------------------------------------------------------------"
+
+    # Delete generated files: CDK Consctruct typescript transcompiled generted files
+    do_cmd cd $source_dir/
+    do_cmd npm run cleanup:tsc
```

Comment:
any reason why we use `npm` here instead of `pnpm` or `rushx`? Where is `cleanup:tsc` defined?

Reply:
replaced with rush command